python/mxnet/executor_manager.py

```diff
@@ -283,8 +283,13 @@ def update_metric(self, metric, labels):
         """ Update evaluation metric with label and current outputs """
         for texec, islice in zip(self.train_execs, self.slices):
             n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))
-            new_slice = slice(islice.start * n, islice.stop * n)
-            labels_slice = [label[new_slice] for label in labels]
+            # special case for ctc loss
+            if len(labels) > 0 and len(labels[0].shape) == 2 \
+               and n > labels[0].shape[1]:
```

Comment:
L285 `n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))` was first introduced as a fix on labels with different batch size from outputs.
We thought if `_split_input_slice` handles first dimension as input slicing, it would be possible to determine a "true" slice for some label with larger batch size than the batch size of data. An example is the ROIPooling symbol.

Try if this works:
```
        for texec, islice in zip(self.train_execs, self.slices):
            n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))
            # special case if label has different batch size
            if n != self.batch_size:
                labels_slice = [label[islice] for label in labels]
            else:
                new_slice = slice(islice.start * n, islice.stop * n)
                labels_slice = [label[new_slice] for label in labels]
            metric.update(labels_slice, texec.outputs)
```